### PR TITLE
Added PyTorch and ONNX Support for inference

### PR DESCRIPTION
**Pull request by boonedoggle**
New Features:
* Created PyTorch version of the average power neural network.
* Added ONNX support
* Added `onnx2plan.py` and updated `trt_utils.py` to support a plan file build from ONNX and UFF.
* Added ONNX exporting from TensorFlow instead of UFF.
* Added Anaconda environment .yml file
* Added better tutorial in the

### CODE REVIEW

File: python/inference/run_airt_inference.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def main():
     sdr.setFrequency(SOAPY_SDR_RX, CHANNEL, CENTER_FREQ)
     rx_stream = sdr.setupStream(SOAPY_SDR_RX, SOAPY_SDR_CF32, [CHANNEL])
     sdr.activateStream(rx_stream)
-
+
```
Comment (dwd-pete) [inline python/inference/run_airt_inference.py]: Whitespace added here unnecessarily.

File: python/inference/run_airt_inference.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def main():
                 continue
             # Run samples through neural network
             dnn.feed_forward()
-
+
```
Comment (dwd-pete) [inline python/inference/run_airt_inference.py]: Whitespace.

File: python/inference/tensorflow/make_avg_pow_net.py [hunk 3]
```diff
@@ -77,18 +74,17 @@ def passed_test(buff, tf_result):
...
                          format(np_result.shape, tf_result.shape))
-    print(np_result[0], tf_result[0])
     return np.allclose(np_result, tf_result)
 
 
 def main():
     # Define input length of neural network. Should be 2 x number of complex samples
     input_len = 2 * CPLX_SAMPLES_PER_INFER
-
+
```
Comment (dwd-pete) [inline python/inference/tensorflow/make_avg_pow_net.py]: Whitespace.

File: python/inference/tensorflow/make_avg_pow_net.py [hunk 4]
```diff
@@ -77,18 +74,17 @@ def passed_test(buff, tf_result):
...
 def main():
     # Define input length of neural network. Should be 2 x number of complex samples
     input_len = 2 * CPLX_SAMPLES_PER_INFER
-
+    
     # Create test data in range of (-1, 1) for forward propagation test
     batch_size = 128  # Batch size to use for testing and validation here
     buff = np.random.randn(batch_size, input_len).astype(np.float32)
-
+
```
Comment (dwd-pete) [inline python/inference/tensorflow/make_avg_pow_net.py]: Whitespace.

File: python/inference/pytorch/make_avg_pow_net.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+        Returns:
+            output: average power tensor, i.e., avg(sig.real^2 + sig.imag^2)
+        """
+        # Create layer that squares each sample independently
+        layer1 = torch.pow(batch_x, 2.0)
+        # Create layer2 that uses a matrix multiplication to multiply each squared
+        # sample by a normalization factor then sums all samples to compute the average
+        # of layer1.
+        layer2 = torch.mm(layer1, self.norm_matrix)
```
Comment (dwd-pete) [inline python/inference/pytorch/make_avg_pow_net.py:76]: The way these layers are created differs from what we have in our internal test scripts. Looking at the API calls, these should result in the same network.

File: python/inference/README.md [hunk 6]
```diff
@@ -49,115 +61,218 @@ performed.
...
-### Train Neural Network
-Currently only TensorFlow is demonstrated with this example code. This process should be 
-run on a computer with TensorFlow installed. Since there is no training involved, any 
-computer with TensorFlow installed will work, i.e., no GPU required. Note that the AIR-T
-does not come preinstalled with TensorFlow, however users may be able to install it on 
-the platform.
+## Step 1: Train Neural Network
+This toolbox provides examples for both TensorFlow and PyTorch. The training process
+should be run on a computer with TensorFlow and/or PyTorch installed. Since there is no
+training involved, any computer with TensorFlow installed will work, i.e., no GPU required.
```
Comment (dwd-pete) [inline python/inference/README.md]: I think you want "TensorFlow and/or PyTorch" in this line.

File: python/inference/README.md [hunk 7]
```diff
@@ -168,5 +283,25 @@ Receiving Data
...
+for AirStack 0.4.0+. For the simplicity of interacting with the AirStack drivers,
+we recommend using the Python Packaging instead of Docker. The Python package should be
+installed in the
+[AirStack Anaconda environment](http://docs.deepwavedigital.com/Tutorials/6_conda.html).
 
-Copyright 2020, Deepwave Digital, Inc.
\ No newline at end of file
+
+## Summary
+This toolobox should serve as an example for performing neural network inference on the AIR-T
```
Comment (dwd-pete) [inline python/inference/README.md]: toolbox misspelled

**Review by dwd-pete — COMMENTED** [review]
Minor clean-up probably needed before merge.

File: python/inference/pytorch/make_avg_pow_net.py (hunk 5 above)
Comment (boonedoggle) [reply python/inference/pytorch/make_avg_pow_net.py:76]: Yes, I thought the TensorFlow and PyTorch examples were more consistent with each other this way. The functionality is the same using `torch.mm(layer1, self.norm_matrix)` is the same as `layer1.mm(self.norm_matrix)`.

**Pull request merged**